src/utils/risk.ts

```diff
@@ -0,0 +1,275 @@
+    const annual_average = avgSafe(monthlyValueList);
+
+    return {
+        ...ipcRiskDataItem,
+        annual_average,
+    };
+}
+
+export function getFiRiskDataItem(data: IpcData | undefined) {
+    if (isNotDefined(data) || data.length === 0) {
+        return undefined;
+    }
+
+    const uniqueData = getPrioritizedIpcData(data);
+    const ipcRiskDataItem = getAverageIpcData(uniqueData);
+    const firstData = data[0];
+
+    return {
+        hazard_type: 'FI' as const,
+        hazard_type_display: firstData?.hazard_type_display,
+        country_details: firstData?.country_details,
+        ...ipcRiskDataItem,
+    };
+}
+
+export function getWfRiskDataItem(data: GwisData | undefined) {
+    if (isNotDefined(data) || data.length === 0) {
+        return undefined;
+    }
+
+    const uniqueData = unique(
+        data,
+        (gwisDataItem) => `${gwisDataItem.year}-${gwisDataItem.month}`,
+    );
+
+    const monthGroupedGwisData = listToGroupList(
+        uniqueData,
+        (datum) => datum.month,
+        (datum) => datum.dsr_avg,
+    );
+
+    const gwisRiskDataItem = mapToMap(
+        monthGroupedGwisData,
+        (monthKey) => monthNumberToNameMap[Number(monthKey) - 1],
+        (item) => avgSafe(item),
+    );
+
+    const firstYearData = data[0];
+    const monthlyValueList = Object.values(gwisRiskDataItem);
+    const annual_average = avgSafe(monthlyValueList);
+
+    return {
+        hazard_type: 'WF' as const,
+        hazard_type_display: firstYearData?.hazard_type_display,
+        country_details: firstYearData?.country_details,
+        ...gwisRiskDataItem,
+        annual_average,
+    };
+}
+
+export function hasSomeDefinedValue(riskDataItem: RiskDataItem) {
+    return mapToList(
+        monthNumberToNameMap,
+        // FIXME: we should avoid !== 0 condition
+        (value) => isDefined(riskDataItem[value]) && riskDataItem[value] !== 0,
+    ).some(Boolean);
+}
+
+export function riskScoreToCategory(score: number | undefined | null, hazardType: HazardType) {
+    if (isNotDefined(score) || score < 0) {
+        return undefined;
+    }
+
+    const scoreCategories = hazardType === 'WF' ? [
+        { score: 1000, category: CATEGORY_RISK_VERY_HIGH },
+        { score: 17, category: CATEGORY_RISK_HIGH },
+        { score: 9, category: CATEGORY_RISK_MEDIUM },
+        { score: 5, category: CATEGORY_RISK_LOW },
+        { score: 2, category: CATEGORY_RISK_VERY_LOW },
+    ] : [
+        { score: 10, category: CATEGORY_RISK_VERY_HIGH },
+        { score: 6.5, category: CATEGORY_RISK_HIGH },
+        { score: 5, category: CATEGORY_RISK_MEDIUM },
+        { score: 3.5, category: CATEGORY_RISK_LOW },
+        { score: 2, category: CATEGORY_RISK_VERY_LOW },
+    ];
+
+    const currentCategory = scoreCategories.reverse().filter(
+        (category) => score <= category.score,
+    );
+
+    if (currentCategory.length === 0) {
+        return undefined;
+    }
+
+    return currentCategory[0].category;
```

Comment:
Why are we using the first data? Why did we reverse?

Reply:
We're trying to find the category for the selected score

Reply:
Let's add NOTE